### DISCUSSION TITLE
I want to know the reason that 'Run' button doesn't show in 'Auto Segmentation'

### DISCUSSION THREAD

**Discussion by ulphypro** [Project-MONAI/MONAILabel#1346] [Q&A]
Can you see 'Run' button in 'Auto Segmentation' option?

I really don't know why it doesn't show 'Run' button in 'Auto Segmentation' option.
![image](https://user-images.githubusercontent.com/67679322/226274982-a2a308ac-81ca-4b9c-8433-8907cf2d8d8b.png)

I already installed torch 1.13.1, torchaudio 0.13.1, and torchvision 0.14.1. 

I also installed monailabel 0.6.0 version.

Because 3D-Slicer only suuported 1 channel, I downloaded MICCAI BraTS 2021 challenge brain tumor dataset instead of Task01_BrainTumour dataset.
[(MICCAI BraTS 2021 challenge)](http://braintumorsegmentation.org/)

A number of t2_nii.gz files include in one folder as following.
![image](https://user-images.githubusercontent.com/67679322/226274437-0cea609e-c3a7-4942-9eb7-16c5f6c2ffeb.png)

To segment brain tumor in image, I modified apps/radiology/lib/config/segmentation.py. :

1.         # Labels
        self.labels = {
            "tumor": 1,
            
        }

2.       # Download PreTrained Model
        if strtobool(self.conf.get("use_pretrained_model", "false")):

And  because it occurred warning messages in 5.2.2 version, I'm using slicer 5.0.3 version. it didn't occur any warning messages in slicer 5.0.3.

When I had used 3D-Slicer 5.2.2, 5.2.1, 5.0.3 and 5.0.2, all of them didn't show 'Run' button in 'Auto Segmentation' option.

Please, let me know how to solve regarding the problem that 'Run' button don't show in 'Auto Segmentation' option based on Linux and Windows 11.

**Comment by SachidanandAlle** [comment]
You can visit in browser to go through all infer and train compatible models if available

http://127.0.0.1:8000/info/

**Reply by ulphypro** [reply]
Dear @SachidanandAlle 


Thank you for answering my question.

I input 'http://127.0.0.1:8000/info/'  in 'monailabel server' option.

And then I pressed 'refresh' button.

But it occurrs error. The error message is 'Failed to fetch models from remote server. Make sure server address is correct and <server_uri>/info/ is accessible in browser'.

Also, error message shown in Python Interactor in 3D-Slicer is as following.

![image](https://user-images.githubusercontent.com/67679322/226411814-116651ff-34c5-4e2c-a648-f8e69a8cc077.png)

I connected 'http://127.0.0.1:8000/info/' using web browser. The result is as following.
![image](https://user-images.githubusercontent.com/67679322/226415799-4dba4bd7-4d69-4e46-8f17-78126ee93052.png)


Please, let me know how to solve the problem.

And if I connect 'http://127.0.0.1:8000/info/' in 'monailabel server' option, Can 'Run' button activate in 'Auto Segmentation' option?

**Reply by SachidanandAlle** [reply]
http://127.0.0.1:8000/info/ in web browser... Not in slicer

I think you r using older slicer or plugin.. I suggest to uninstall and try new one

**Reply by ulphypro** [reply]
Dear @SachidanandAlle 

Despite I already uninstalled and reinstalled 3D-Slicer 5.2.2 version and monailabel plugin, I didn't show 'Run' button in 'Auto Segmentation' option. Refer to #1314.

Can you see 'Run' button in 'Auto Segmentation' option when you run monailabel server?

I didn't solve the problem that is deactivation of 'Run' button in 'Auto Segmentation' option during several weeks.

I know that you modified MONAILabel module yesterday. If I reinstall Monailabel module and run monailabel server, can 'Run' button activate in 'Auto segmentation' option?

**Reply by diazandr3s** [reply]
Hi @ulphypro,

It is great to see you're getting a good DICE score on the BRATS dataset using a single modality (T2).
It is a bit strange that the Run button is grey. Is there a model listed in the Auto-Segmentation dropdown menu?

**Reply by ulphypro** [reply]
Dear @diazandr3s 

Thank you for answering my question.

There is no model listed when I opened dropdown menu of 'Auto Segmentation'. 



I often experienced this problem regarding deactivation of 'Run button' in 'Auto Segmentation' option.

Whenever It occurs the problem. I used to reinstall new version MONAILabel and 3D-Slicer software.  Some days 'Run' buttion used to show well.  

Now, despite I reinstalled new version of monailabel and 3D-Slicer, I couldn't show 'Run' button in 'Auto Segmentation' option.

I'm using 3D-Slicer 5.2.2, torch 1.13.1 and monailabel 0.6.0.

When you run MONAILabel server in 3D-Slicer, can you see 'Run' button in 'Auto Segmentation' option well?
 
Please, help me to solve the problem.

**Reply by SachidanandAlle** [reply]
Can you provide the json output from the browser for the URI I mentioned

**Reply by ulphypro** [reply]
Dear @SachidanandAlle

Result connected 'http://127.0.0.1:8000/info/' in web browser is as following.

{"name":"MONAILabel - Radiology (0.6.0)","description":"DeepLearning models for radiology","version":"0.6.0","labels":["tumor"],"models":{"segmentation":{"type":"segmentation","labels":{"tumor":1},"dimension":3,"description":"A pre-trained model for volumetric (3D) Segmentation from CT image","config":{"largest_cc":false,"device":["cuda"]}},"Histogram+GraphCut":{"type":"scribbles","labels":{"tumor":1},"dimension":3,"description":"A post processing step with histogram-based GraphCut for Generic segmentation","config":{"num_bins":64,"lamda":1.0,"sigma":0.1,"device":["cuda"]}},"GMM+GraphCut":{"type":"scribbles","labels":{"tumor":1},"dimension":3,"description":"A post processing step with GMM-based GraphCut for Generic segmentation","config":{"num_mixtures":20,"lamda":5.0,"sigma":0.5,"device":["cuda"]}}},"trainers":{"segmentation":{"description":"Train Segmentation Model","config":{"name":"train_01","pretrained":true,"device":"cuda","max_epochs":50,"early_stop_patience":-1,"val_split":0.2,"train_batch_size":1,"val_batch_size":1,"multi_gpu":true,"gpus":"all","dataset":["SmartCacheDataset","CacheDataset","PersistentDataset","Dataset"],"dataloader":["ThreadDataLoader","DataLoader"],"tracking":["mlflow","None"],"tracking_uri":"","tracking_experiment_name":""},"labels":{"tumor":1}}},"strategies":{"random":{"description":"Random Strategy"},"first":{"description":"Get First Sample"},"last":{"description":"Get Last Sample"}},"scoring":{},"train_stats":{"segmentation":{"rank":0,"current_epoch":100,"current_iteration":2800,"total_epochs":100,"total_iterations":28,"epoch":100,"start_ts":1679015154,"total_time":"1:02:48","best_metric":0.8132548928260803,"train":{"metrics":{"train_mean_dice":0.8089019060134888,"train_tumor_mean_dice":0.8089019060134888},"key_metric_name":"train_mean_dice","best_metric":0.8864032626152039,"best_metric_epoch":54},"eval":{"metrics":{"val_mean_dice":0.7993720173835754,"val_tumor_mean_dice":0.7993720173835754},"key_metric_name":"val_mean_dice","best_metric":0.8132548928260803,"best_metric_epoch":43}}},"datastore":{"total":119,"completed":35,"label_tags":{"final":35}}}

**Comment by SachidanandAlle** [comment]
This says there is no issue with monailabel server.. you can run under/segmentation on an image..

Have you tried pre release of slicer version instead?

Or you can uninstall monailabel plugin and configure it development mode..

**Reply by ulphypro** [reply]
Dear @SachidanandAlle 


It seems that the reason that didn't show 'Run' button in Auto Segmentation is systemic problem in 3D Slicer. Is it right?

I had never been to use pre release of slicer.

Now I want to install pre release of slicer(5.3.0), but download site of 3D Slicer occurs internar server error.

Does 'Run' button activate when i reinstall pre release of 3D slicer,  uninstall monailabel plugin or configure it development mode?

I should use Auto Segmentation using monailabel server in 3D Slicer.

Please let me know in detail what I should do to activate the 'Run' button in 'Auto Segmentation' option.

**Reply by ulphypro** [reply]
Now, I installed pre release of 3D Slicer updated today, and then I ran 3D Slicer. 

 'Run' button shows in Auto Segmentation option based on 3D Slicer 5.3.0 version. 

@SachidanandAlle  Thank you for helping me.
